src/local-ssh/proxy.ts

```diff
@@ -243,7 +249,10 @@ class WebSocketSSHProxy {
 
     async retryGetWorkspaceInfo(username: string) {
         return retry(async () => {
-            return this.extensionIpc.getWorkspaceAuthInfo({ workspaceId: username, gitpodHost: this.options.host }).catch(e => {
+            const controller = new AbortController();
+            const signal = controller.signal;
+            setTimeout(() => { controller.abort(); }, 10000);
```

Comment:
> it is already bound to Remote SSH timeout and we should respect it.

But it doesn't work for all our old release, I add it in case listen to `close` and `end` is not enough

Reply:
It's not harmful for our case, and may reduce requests if something wrong

Reply:
I don't think I understand sorry. So we have a remote window it starts a proxy and tries to connect for some timeout. Here we do a request which will hang till the process is destroyed by Remote SSH timeout. When a proxy process is destroyed the abort signal is triggered in the extension server. We checked it with Public API and it was working.